### PR TITLE
Themr breaks shouldComponentUpdate shallow equal optimization

### PR DESCRIPTION
**Pull request by istarkov**
Add failing test showing #25

### CODE REVIEW

**Comment by istarkov** [comment]
Add fix.

**Comment by istarkov** [comment]
`ThemeProvider enforces a single child` error was before my commit.

File: src/components/themr.js [hunk 1]
```diff
@@ -80,30 +85,49 @@ export default (componentName, localTheme, options = {}) => (ThemedComponent) =>
...
+    }
+
+    calcTheme(props) {
+      const { composeTheme } = props
+      return composeTheme
+        ? this.getTheme(props)
+        : this.getThemeNotComposed(props)
+    }
+
+    componentWillReceiveProps(nextProps) {
```
Comment (raveclassic) [inline src/components/themr.js]: How do you handle context here? Seems like component won't react on changing its theme in context. Even without pure-rendering optimization.

Comment (raveclassic) [inline src/components/themr.js]: After breif dicussion with @istarkov it seems that `ThemeProvider` should have some kind of notification mechanism to notify themed components about context changes. This is out of scope of this PR.

**Comment by raveclassic** [comment]
Here's a [quickfix](https://github.com/javivelasco/react-css-themr/pull/24#issuecomment-249566727)
